packages/amazonq/src/extensionShared.ts

```diff
@@ -156,16 +156,18 @@ export async function activateShared(context: vscode.ExtensionContext, isWeb: bo
             telemetry.record({ source: ExtStartUpSources.reload })
         }
 
-        const authState = (await AuthUtil.instance.getChatAuthState()).codewhispererChat
         const authKinds: AuthUtils.AuthSimpleId[] = []
         if (await AuthUtils.hasBuilderId('codewhisperer')) {
             authKinds.push('builderIdCodeWhisperer')
         }
         if (await AuthUtils.hasSso('codewhisperer')) {
             authKinds.push('identityCenterCodeWhisperer')
         }
```

Comment:
this section is only relevant if we did not use the authStatus of the autoconnection. Maybe getAuthStatus can also do this work for both the AuthUtils (if it has it) or the autoconnection. This function will do the work for you, so it should be an easy change: https://github.com/aws/aws-toolkit-vscode/blob/feature/standalone/packages/core/src/login/webview/vue/backend.ts#L280

Reply:
nevermind, this isn't quite the same metric as what the function provides. we will have to come back to this.

Reply:
This PR will add an appropriate util function. We can follow up. https://github.com/aws/aws-toolkit-vscode/pull/4860